src/IoHandlers.h

```diff
@@ -373,6 +375,94 @@ class IoEpoll: public IoHandler {
 	int m_max_events;
 };
 #endif
-#endif
 
+#ifdef  USING_VMA_EXTRA_API
+//==============================================================================
+class IoVmaPoll: public IoHandler {
+public:
+	IoVmaPoll(int _fd_min, int _fd_max, int _fd_num);
+	virtual ~IoVmaPoll();
+
+	//------------------------------------------------------------------------------
+	inline void update() {
+		int ifd = 0;
+		
+		m_look_start = 0;
+		m_look_end = m_fd_num;
+		for (ifd = m_fd_min; ifd <= m_fd_max; ifd++ ) {
+			if (g_fds_array[ifd]) {
+				int i = 0;
+				int active_fd_count = g_fds_array[ifd]->active_fd_count;
+				int *active_fd_list = g_fds_array[ifd]->active_fd_list;
+
+				assert( active_fd_list &&
+					"corrupted fds_data object" );
+
+				while (active_fd_count) {
+					/* process active sockets in case TCP (listen sockets are set in prepareNetwork()) and
+					 * skip active socket in case UDP (it is the same with set in prepareNetwork())
+					 */
+					if (active_fd_list[i] != (int)INVALID_SOCKET) {
+						active_fd_count--;
+					}
+					i++;
+					
+					assert((i < MAX_ACTIVE_FD_NUM) &&
+					       "maximum number of active connection to the single TCP addr:port");
+				}
+			}
+		}
+        }
+	//------------------------------------------------------------------------------
+	inline int waitArrival(){
+		m_look_end = 0;
+		for ( m_rings_vma_comps_map_itr = m_rings_vma_comps_map.begin(); m_rings_vma_comps_map_itr != m_rings_vma_comps_map.end(); ++m_rings_vma_comps_map_itr ){
```

Comment:
please consider changing `for ( m_rings_vma_comps_map_itr` with `for (m_rings_vma_comps_map_itr` (w/o space)